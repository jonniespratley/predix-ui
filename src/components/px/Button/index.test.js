import { expect } from 'chai';
import React from 'react';
import {shallow} from 'enzyme';
import PxButton from './';

describe('Button', () => {
  test('should...', () =>{
    const wrapper = shallow(
      <PxButton/>
    );
    console.log(wrapper.debug());
    expect(true).to.equal(true);
  });
  test('should render button', () => {
    const wrapper = shallow(<PxButton>Button</PxButton>);
    console.log(wrapper.debug());
    //expect(wrapper.find('.btn')).to.have.length(1);
  });
  test('should render label and style', () => {
    const wrapper = shallow(<PxButton primary>Button</PxButton>);
    console.log(wrapper.debug());
    //expect(wrapper.find('.btn')).to.have.length(1);
    //expect(wrapper.find('.btn--primary')).to.have.length(1);
  });
});
