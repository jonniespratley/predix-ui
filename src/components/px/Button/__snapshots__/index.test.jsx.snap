// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render bare button 1`] = `
.c0 {
  font-family: "GE Inspira Sans";
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  overflow: visible;
  height: var(--px-btn-height,auto);
  min-height: var(--px-btn-min-height,2rem);
  min-width: var(--px-btn-min-width,4.66667em);
  margin: 0;
  border: 1px solid var(--px-btn-border-color,transparent);
  border-radius: 0 !important;
  padding: calc(var(--px-btn-height,2em) / 10) calc(var(--px-btn-height,2em) / 2);
  box-shadow: var(--px-btn-shadow--light,none);
  font: inherit;
  line-height: calc(var(--px-btn-height,2em) - 2px);
  cursor: pointer;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  white-space: nowrap;
  background-color: var(--px-btn-background,#d8e0e5);
  -webkit-transition: background .4s,border-color .4s,color .4s;
  transition: background .4s,border-color .4s,color .4s;
  -webkit-appearance: button;
  -webkit-font-smoothing: antialiased;
  color: var(--px-btn-color,#2c404c);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0 !important;
  border-radius: 0 !important;
  line-height: inherit;
  padding-left: 0;
  padding-right: 0;
  box-shadow: none;
  background: none;
  outline: none;
}

.c0:link,
.c0:visited,
.c0:hover,
.c0:active {
  color: var(--px-btn-color,#2c404c);
}

.c0:hover,
.c0:focus {
  border-color: var(--px-btn-border-color--hover,transparent);
  box-shadow: var(--px-btn-shadow,none);
  background-color: var(--px-btn-background--hover,#a3b5bf);
}

.c0:active {
  outline: none;
  border-color: var(--px-btn-border-color--pressed,transparent);
  box-shadow: none;
  background-color: var(--px-btn-background--pressed,#889aa5);
}

.c0:link,
.c0:visited {
  box-shadow: none;
  background: none;
  outline: none;
  color: var(--px-btn-bare-color,#2c404c);
}

.c0:hover,
.c0:focus {
  color: var(--px-btn-bare-color--hover,#007acc);
  background-color: transparent;
}

.c0:active {
  color: var(--px-btn-bare-color--pressed,#003d66);
  background-color: transparent;
}

<button
  className="px-button c0"
  icon={null}
  size={null}
  type={null}
>
  Button
</button>
`;

exports[`Button should render bare-primary button 1`] = `
.c0 {
  font-family: "GE Inspira Sans";
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  overflow: visible;
  height: var(--px-btn-height,auto);
  min-height: var(--px-btn-min-height,2rem);
  min-width: var(--px-btn-min-width,4.66667em);
  margin: 0;
  border: 1px solid var(--px-btn-border-color,transparent);
  border-radius: 0 !important;
  padding: calc(var(--px-btn-height,2em) / 10) calc(var(--px-btn-height,2em) / 2);
  box-shadow: var(--px-btn-shadow--light,none);
  font: inherit;
  line-height: calc(var(--px-btn-height,2em) - 2px);
  cursor: pointer;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  white-space: nowrap;
  background-color: var(--px-btn-background,#d8e0e5);
  -webkit-transition: background .4s,border-color .4s,color .4s;
  transition: background .4s,border-color .4s,color .4s;
  -webkit-appearance: button;
  -webkit-font-smoothing: antialiased;
  color: var(--px-btn-color,#2c404c);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0 !important;
  border-radius: 0 !important;
  line-height: inherit;
  padding-left: 0;
  padding-right: 0;
  box-shadow: none;
  background: none;
  outline: none;
  color: var(--px-btn-primary-bare-color--visited,#007acc);
}

.c0:link,
.c0:visited,
.c0:hover,
.c0:active {
  color: var(--px-btn-color,#2c404c);
}

.c0:hover,
.c0:focus {
  border-color: var(--px-btn-border-color--hover,transparent);
  box-shadow: var(--px-btn-shadow,none);
  background-color: var(--px-btn-background--hover,#a3b5bf);
}

.c0:active {
  outline: none;
  border-color: var(--px-btn-border-color--pressed,transparent);
  box-shadow: none;
  background-color: var(--px-btn-background--pressed,#889aa5);
}

.c0:disabled {
  color: var(--px-btn-disabled-color,rgba(0,0,0,0.2));
}

.c0:link,
.c0:visited,
.c0:hover,
.c0:active {
  box-shadow: none;
  background: none;
  outline: none;
}

.c0:hover,
.c0:focus {
  color: var(--px-btn-primary-bare-color--hover,#005c99);
}

.c0:active,
:focus {
  color: var(--px-btn-primary-bare-color--pressed,#003d66);
}

<button
  className="px-button c0"
  icon={null}
  size={null}
  type={null}
>
  Button
</button>
`;

exports[`Button should render call-to-action button 1`] = `
.c0 {
  font-family: "GE Inspira Sans";
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  overflow: visible;
  height: var(--px-btn-height,auto);
  min-height: var(--px-btn-min-height,2rem);
  min-width: var(--px-btn-min-width,4.66667em);
  margin: 0;
  border: 1px solid var(--px-btn-border-color,transparent);
  border-radius: 0 !important;
  padding: calc(var(--px-btn-height,2em) / 10) calc(var(--px-btn-height,2em) / 2);
  box-shadow: var(--px-btn-shadow--light,none);
  font: inherit;
  line-height: calc(var(--px-btn-height,2em) - 2px);
  cursor: pointer;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  white-space: nowrap;
  background-color: var(--px-btn-background,#d8e0e5);
  -webkit-transition: background .4s,border-color .4s,color .4s;
  transition: background .4s,border-color .4s,color .4s;
  -webkit-appearance: button;
  -webkit-font-smoothing: antialiased;
  color: var(--px-btn-color,#2c404c);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--px-btn-call-to-action-color,white);
  border-color: var(--px-btn-call-to-action-border-color,transparent);
  box-shadow: var(--px-btn-shadow,none);
  background-color: var(--px-btn-call-to-action-background,#007acc);
}

.c0:link,
.c0:visited,
.c0:hover,
.c0:active {
  color: var(--px-btn-color,#2c404c);
}

.c0:hover,
.c0:focus {
  border-color: var(--px-btn-border-color--hover,transparent);
  box-shadow: var(--px-btn-shadow,none);
  background-color: var(--px-btn-background--hover,#a3b5bf);
}

.c0:active {
  outline: none;
  border-color: var(--px-btn-border-color--pressed,transparent);
  box-shadow: none;
  background-color: var(--px-btn-background--pressed,#889aa5);
}

.c0:link,
.c0:visited,
.c0:hover,
.c0:active {
  color: var(--px-btn-call-to-action-color,white);
}

.c0:hover,
.c0:focus {
  background-color: var(--px-btn-call-to-action-background--hover,#005c99);
}

.c0:active {
  background-color: var(--px-btn-call-to-action-background--pressed,#003d66);
}

<button
  className="px-button c0"
  icon={null}
  size={null}
  type={null}
>
  Button
</button>
`;

exports[`Button should render primary button 1`] = `
.c0 {
  font-family: "GE Inspira Sans";
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  overflow: visible;
  height: var(--px-btn-height,auto);
  min-height: var(--px-btn-min-height,2rem);
  min-width: var(--px-btn-min-width,4.66667em);
  margin: 0;
  border: 1px solid var(--px-btn-border-color,transparent);
  border-radius: 0 !important;
  padding: calc(var(--px-btn-height,2em) / 10) calc(var(--px-btn-height,2em) / 2);
  box-shadow: var(--px-btn-shadow--light,none);
  font: inherit;
  line-height: calc(var(--px-btn-height,2em) - 2px);
  cursor: pointer;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  white-space: nowrap;
  background-color: var(--px-btn-background,#d8e0e5);
  -webkit-transition: background .4s,border-color .4s,color .4s;
  transition: background .4s,border-color .4s,color .4s;
  -webkit-appearance: button;
  -webkit-font-smoothing: antialiased;
  color: var(--px-btn-color,#2c404c);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--px-btn-primary-color,white);
  border-color: var(--px-btn-primary-border-color,transparent);
  box-shadow: var(--px-btn-shadow,none);
  background-color: var(--px-btn-primary-background,#364c59);
}

.c0:link,
.c0:visited,
.c0:hover,
.c0:active {
  color: var(--px-btn-color,#2c404c);
}

.c0:hover,
.c0:focus {
  border-color: var(--px-btn-border-color--hover,transparent);
  box-shadow: var(--px-btn-shadow,none);
  background-color: var(--px-btn-background--hover,#a3b5bf);
}

.c0:active {
  outline: none;
  border-color: var(--px-btn-border-color--pressed,transparent);
  box-shadow: none;
  background-color: var(--px-btn-background--pressed,#889aa5);
}

.c0:link,
.c0:visited,
.c0:hover,
.c0:active {
  color: var(--px-btn-primary-color,white);
}

.c0:hover,
.c0:focus {
  border-color: transparent;
  background-color: var(--px-btn-primary-background--hover,#23343f);
}

.c0:active {
  border-color: transparent;
  background-color: var(--px-btn-primary-background--pressed,#121f26);
}

<button
  className="px-button c0"
  icon={null}
  size={null}
  type={null}
>
  Button
</button>
`;

exports[`Button should render secondary button 1`] = `
.c0 {
  font-family: "GE Inspira Sans";
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  overflow: visible;
  height: var(--px-btn-height,auto);
  min-height: var(--px-btn-min-height,2rem);
  min-width: var(--px-btn-min-width,4.66667em);
  margin: 0;
  border: 1px solid var(--px-btn-border-color,transparent);
  border-radius: 0 !important;
  padding: calc(var(--px-btn-height,2em) / 10) calc(var(--px-btn-height,2em) / 2);
  box-shadow: var(--px-btn-shadow--light,none);
  font: inherit;
  line-height: calc(var(--px-btn-height,2em) - 2px);
  cursor: pointer;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  white-space: nowrap;
  background-color: var(--px-btn-background,#d8e0e5);
  -webkit-transition: background .4s,border-color .4s,color .4s;
  transition: background .4s,border-color .4s,color .4s;
  -webkit-appearance: button;
  -webkit-font-smoothing: antialiased;
  color: var(--px-btn-color,#2c404c);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:link,
.c0:visited,
.c0:hover,
.c0:active {
  color: var(--px-btn-color,#2c404c);
}

.c0:hover,
.c0:focus {
  border-color: var(--px-btn-border-color--hover,transparent);
  box-shadow: var(--px-btn-shadow,none);
  background-color: var(--px-btn-background--hover,#a3b5bf);
}

.c0:active {
  outline: none;
  border-color: var(--px-btn-border-color--pressed,transparent);
  box-shadow: none;
  background-color: var(--px-btn-background--pressed,#889aa5);
}

<button
  className="px-button c0"
  icon={null}
  size={null}
  type={null}
>
  Button
</button>
`;

exports[`Button should render tertiary button 1`] = `
.c0 {
  font-family: "GE Inspira Sans";
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  overflow: visible;
  height: var(--px-btn-height,auto);
  min-height: var(--px-btn-min-height,2rem);
  min-width: var(--px-btn-min-width,4.66667em);
  margin: 0;
  border: 1px solid var(--px-btn-border-color,transparent);
  border-radius: 0 !important;
  padding: calc(var(--px-btn-height,2em) / 10) calc(var(--px-btn-height,2em) / 2);
  box-shadow: var(--px-btn-shadow--light,none);
  font: inherit;
  line-height: calc(var(--px-btn-height,2em) - 2px);
  cursor: pointer;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  white-space: nowrap;
  background-color: var(--px-btn-background,#d8e0e5);
  -webkit-transition: background .4s,border-color .4s,color .4s;
  transition: background .4s,border-color .4s,color .4s;
  -webkit-appearance: button;
  -webkit-font-smoothing: antialiased;
  color: var(--px-btn-color,#2c404c);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-color: var(--px-btn-tertiary-border-color,#889aa5);
  border: 1px solid;
  box-shadow: none;
  background-color: var(--px-btn-tertiary-background,white);
}

.c0:link,
.c0:visited,
.c0:hover,
.c0:active {
  color: var(--px-btn-color,#2c404c);
}

.c0:hover,
.c0:focus {
  border-color: var(--px-btn-border-color--hover,transparent);
  box-shadow: var(--px-btn-shadow,none);
  background-color: var(--px-btn-background--hover,#a3b5bf);
}

.c0:active {
  outline: none;
  border-color: var(--px-btn-border-color--pressed,transparent);
  box-shadow: none;
  background-color: var(--px-btn-background--pressed,#889aa5);
}

.c0:link,
.c0:visited,
.c0:hover,
.c0:active {
  border-color: var(--px-btn-tertiary-border-color,#889aa5);
  color: var(--px-btn-tertiary-color,#007acc);
}

.c0:hover,
.c0:focus {
  color: var(--px-btn-tertiary-color--hover,#005c99);
  border-color: var(--px-btn-tertiary-border-color,#889aa5);
  box-shadow: none;
  background-color: var(--px-btn-tertiary-background--hover,#c5d1d8);
}

.c0:active {
  color: var(--px-btn-tertiary-color--pressed,#003d66);
  border-color: var(--px-btn-tertiary-border-color,#889aa5);
  box-shadow: none;
  background-color: var(--px-btn-tertiary-background--pressed,#96a8b2);
}

<button
  className="px-button c0"
  icon={null}
  size={null}
  type={null}
>
  Button
</button>
`;
